src/librustc_mir/borrow_check/nll/universal_regions.rs

```diff
@@ -882,3 +929,23 @@ impl<'tcx> FreeRegionRelations<'tcx> for UniversalRegions<'tcx> {
         self.outlives(longer, shorter)
     }
 }
+
+fn fold_late_bound_regions<'tcx>(
```

Comment:
Nit: I would call this `for_each_late_bound_region` -- `fold` suggests to me that we are propagating a result back. I'd probably even call it `for_each_late_bound_region_defined_on` but I sort of have a penchant for long names. ;)

Reply:
Also a comment is a good idea, something like:

---

Iterates over the late-bound regions defined on `fn_def_id` and invokes `` with the liberated form of each one.